### PR TITLE
General cardinality feature updates

### PR DESCRIPTION
**Pull request by mpsonntag**
This PR is a preparation for the introduction of Section-section and Section-property cardinality features to keep PRs at a size that is still convenient to review.

The PR introduces
- a generalization of the cardinality function for later reuse by different cardinalities
- catches some edge cases in the core cardinality function
- filters the validation warnings when checking for cardinality violations to display only warning messages specific to the validated object and not from connected objects as well.
- adds and updates tests where required

### CODE REVIEW

File: odml/util.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    # Catch tuple edge cases (0, 0); (None, None); (0, None); (None, 0)
+    if isinstance(in_val, tuple) and len(in_val) > 1 and not in_val[0] and not in_val[1]:
+        return None
+
+    # Providing a single integer sets the maximum value in a tuple.
+    if isinstance(in_val, int) and in_val > 0:
+        return None, in_val
+
+    # Only integer 2-tuples of the format '(min, max)' are supported to set the cardinality
+    if isinstance(in_val, tuple) and len(in_val) == 2:
```
Comment (jgrewe) [inline odml/util.py:42]: do we want to allow lists of two values as well? I think one could easily catch this by 
converting `in_val` to `tuple`
